src/export.rs

```diff
@@ -0,0 +1,235 @@
+//! Export support for mmj2 proof files.
+
+use diag::Diagnostic;
+use nameck::Nameset;
+use parser::StatementRef;
+use parser::StatementType;
+use parser::TokenRef;
+use proof::ProofTreeArray;
+use regex::Regex;
+use scopeck::ScopeResult;
+use segment_set::SegmentSet;
+use std::cmp::Ord;
+use std::cmp::Ordering;
+use std::cmp::PartialOrd;
+use std::collections::BinaryHeap;
+use std::error;
+use std::fmt;
+use std::io;
+use std::io::Write;
+use std::str;
+use std::u16;
+
+/// The error type for export::export_mmp().
```

Comment:
You know, Rust has a surefire anti-boilerplate defense mechanism: macros. I wouldn't recommend a wholly different approach to errors because this is Rust standard stuff, but using macros to clean it up is possible if it is repeated enough times to make the similarities obvious. (Yes, I say this knowing your views on macros in general.)